src/Pages/Overview/Overview.js

```diff
@@ -1,7 +1,150 @@
 import React, { Component } from 'react'
+import './Overview.css'
+import axios from 'axios'
+import OverviewProfile from './OverviewProfile.js'
+
+export default class OverviewBoard extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      // All users array, update by callDatabase
+      users: [],
+      search: ''
+    }
+  }
+
+  componentDidMount () {
+    const token = window.localStorage
+      ? window.localStorage.getItem('jwtToken')
+      : ''
+
+    // Immediately direct to /login if no jwtToken token present
+    if (!token) {
+      if (this.props.history) this.props.history.push('/login')
+      return
+    }
+
+    // verify that user is login
+    axios
+      .post('/api/user/verify', { token })
+      .then(res => {
+        this.setState(
+          {
+            currentUser: res.data.email,
+            currentUserLevel: res.data.accessLevel,
+            authToken: token
+          },
+          () => {
+            this.callDatabase()
+          }
+        )
+      })
+      .catch(() => {
+        if (this.props.history) this.props.history.push('/login')
+      })
+  }
+
+  callDatabase () {
+    axios
+      // get all user!
+      .post('/api/user/users', {
+        // don't need email
+        token: this.state.authToken
+      })
+      .then(result => {
+        if (result.status >= 200 && result.status < 300) {
+          this.setState({ users: result.data })
+        }
+      })
+      .catch(err => {
+        console.log(err)
+      })
+  }
+
+  updateSearch (event) {
+    this.setState({ search: event.target.value })
+  }
+
+  /*
+  Delete api
+  parameter: Json object of object to be deleted
+  */
+  deleteUser (user) {
+    // if they decide to delete themselves: delete->logout
+    axios
+      .post('/api/user/delete', {
+        token: this.state.authToken,
+        email: user.email
+      })
+      .then(result => {
+        this.callDatabase() // reload database
+      })
+      .catch(err => {
+        console.log(err)
+      })
+    if (user.email === this.state.currentUser) {
+      // logout
+      window.localStorage.removeItem('jwtToken')
+      window.location.reload()
+      return window.alert('Self-deprecation is an art')
+    }
+  }
+
+  // simply filter array by name
+  search () {
```

Comment:
i understand that this function either returns:

1. all the users if no query present
2. if a query exists, show any results by their first name

the function name `search()` is misleading to a future reader of this. A suggestion I have would be:

1. Have three state variables, `users`, `query`, and `queryResult` (you already have the first two)
    - `users`, holds all the users, `query` holds the query string, and `queryResult` will hold the filtered result of users.
1. Inside of the `<tbody>` on `Line 132`, you can do
```jsx
{this.state.query ?
    /* return an array.map of this.state.queryResult since there is no query */
   : /* return an array.map of this.state.users since there is no query*/ 
}
```
1. Lastly, to update `this.state.queryResult`, change your `updateSearch` function to:
```js
updateSearch (event) {
    const {value} = event.target
  if (!value) return
  const queryResult = this.state.users.filter(data =>
        data.firstName.toLowerCase().includes(value)
    )
  this.setState({ queryResult })
}
```